src/lib/types/src/Flow/Types/Type/AutoCaster.php

```diff
@@ -36,24 +34,15 @@ public function cast(mixed $value) : mixed
      */
     private function castArray(array $value) : array
```

Comment:
Uhm I think I messed up a bit this `castArray`. 

The only thing it does now is that it narrows array types. 

For example when we have array `[1, 2, 3.0]` in which we have int's and float, it will narrow the type to `float` as all `int` can be represented by `float` however I don't think this should be responsibility of a class called `AutoCaster` 😅
Probably I was trying to fix some bug and I did it here without properly thinking this through 🤦‍♂️

I think this is much bigger issue that will need to be solved in separated PR (probably will require some more explicit array types narrowing mechanism) but for now removing key types makes sense.